### ISSUE TITLE
Feature request: character count under/near “compose” textarea

### ISSUE THREAD

**Issue by aviflax** [jointakahe/takahe#146]
I find it inconvenient to not “know” as I’m composing how close I am to the configured post character limit.

My Mac has some tools set up that I could use (“services”) to check the character count, but sometimes I forget, and it’s a bit awkward. Same on my phone, I have a way to check the character count before submitting the form, but it’s cumbersome.

---

A few related ancillary feature requests:

- [ ] if a draft post exceeds the configured character limit, use client-side HTML form validation (`maxlength`) to prevent the form from being submitted when it’s invalid (faster feedback loop, less server load, works offline, etc, etc)
- [ ] if the compose form includes an image attachment, and it’s submitted, and is found to be invalid because the post has too many characters (or for any other reason) let’s not lose that image attachment; let’s retain it (this could be reframed as a bug, I suppose)

(I’ve made the above list a checklist because I suspect that might make it easier to break them out into discrete issues, if so desired.)

---

These are all things I personally could probably contribute, but I don’t have time at this moment, so I wanted to at least just jot these thoughts down and share them.

**Comment by andrewgodwin** [comment]
Agreed - maxlength should be trivial to feed into the form on the Django side, and I'm sure we can put a touch of scripting in to do a character count.

**Comment by shuuji3** [comment]
If you haven't implemented it yet, I think I can help implement the character counter part in hyperscript.

My current idea is similar/same one as Mastodon like this:
- show the remaining character count as you type
- if the count is over the limit
  - change the counter font color to red
  - disable the Post button

Here are the ones on mastodon.social:
<img width="213" alt="Screenshot 2022-12-09 at 0 31 58" src="https://user-images.githubusercontent.com/1425259/206489126-0bd43a24-a258-4376-ae14-5ddb64fa68b3.png"> <img width="193" alt="Screenshot 2022-12-09 at 0 32 26" src="https://user-images.githubusercontent.com/1425259/206489134-5a7ac904-7e63-4470-b8f9-089a8675cdd8.png">

**Comment by andrewgodwin** [comment]
Sure, go for it! Don't worry about perfect styling of it, we can work on that separately if needed.
